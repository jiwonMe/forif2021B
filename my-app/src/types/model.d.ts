import { DocumentReference } from "@firebase/firestore/dist/lite";
export interface CardModel {
    createdDate: number,
    contentType: 'text' | 'link' | 'image',
    src?: string,
    href?: string,
    content: string,
}