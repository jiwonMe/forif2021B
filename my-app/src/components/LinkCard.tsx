import React from 'react'
import { CardModel } from '../types/model'
import { Card } from 'react-bootstrap'

interface CardProp {
    id: string,
    data: CardModel,
    deleteCard: (id: string) => void,
}

const LinkCard: React.FC<CardProp> = ( props ) => {
    const date = new Date()
    date.setTime(props.data.createdDate)

    const { data: card } = props;

    return (
        <div>
            <Card>
                <Card.Img variant="top" src=""/>
                <Card.Body>
                    <Card.Link href={card.href}>{card.content}</Card.Link>
                </Card.Body>
                <Card.Footer className='text-muted'>
                        created at {date.getTime()}
                    </Card.Footer>
            </Card>
        </div>
    )
}

export default LinkCard