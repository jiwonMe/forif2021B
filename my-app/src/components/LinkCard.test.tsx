import React from 'react';
import LinkCard from './LinkCard';
import { render } from '@testing-library/react';
import { CardModel } from '../types/model';

it('shows the props correctly', () => {
    const cardData: CardModel = {
        createdDate: 1482363367071,
        contentType: 'text',
        href: 'https://sketchsoft3d.com/',
        content: 'hello',
    }
    const card = render(
        <LinkCard data={cardData} id={'123'} deleteCard={jest.fn()}></LinkCard>
    )
    
    expect(card.container).toMatchSnapshot()

    card.getByText('hello');
    expect(card.container.querySelector('a')?.href).toBe(cardData.href);
})