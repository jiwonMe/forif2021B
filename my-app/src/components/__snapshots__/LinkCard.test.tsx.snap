// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows the props correctly 1`] = `
<div>
  <div>
    <div
      class="card"
    >
      <img
        class="card-img-top"
        src=""
      />
      <div
        class="card-body"
      >
        <a
          class="card-link"
          href="https://sketchsoft3d.com/"
        >
          hello
        </a>
      </div>
      <div
        class="text-muted card-footer"
      >
        created at 
        1482363367071
      </div>
    </div>
  </div>
</div>
`;
